Error in "Check for New Comments"

First off - thanks for creating this integration, it'll be super helpful (when I can get it running!).

I'm about 99% sure I have everything set up correctly (have the two secrets created, the LinkedIn developer app, and the Heroku instance all set up). However, in the "Check for New Comments" workflow, I hit the following error. If anyone has any ideas I would appreciate the help!

```
Run linkedin_orbit --check-comments
  linkedin_orbit --check-comments
  shell: /usr/bin/bash -e {0}
  env:
    LINKEDIN_ORGANIZATION: ***
    LINKEDIN_TOKEN: ***
    ORBIT_API_KEY: ***
    ORBIT_WORKSPACE_ID: ***
Checking for new LinkedIn comments and posting them to your Orbit workspace...
/opt/hostedtoolcache/Ruby/2.7.2/x64/lib/ruby/gems/2.7.0/gems/linkedin_orbit-0.1.3/lib/linkedin_orbit/linkedin.rb:51:in `get_posts': undefined method `each' for nil:NilClass (NoMethodError)
	from /opt/hostedtoolcache/Ruby/2.7.2/x64/lib/ruby/gems/2.7.0/gems/linkedin_orbit-0.1.3/lib/linkedin_orbit/linkedin.rb:13:in `process_comments'
	from /opt/hostedtoolcache/Ruby/2.7.2/x64/lib/ruby/gems/2.7.0/gems/linkedin_orbit-0.1.3/lib/linkedin_orbit/client.rb:58:in `comments'
	from /opt/hostedtoolcache/Ruby/2.7.2/x64/lib/ruby/gems/2.7.0/gems/linkedin_orbit-0.1.3/scripts/check_comments.rb:13:in `render'
	from /opt/hostedtoolcache/Ruby/2.7.2/x64/lib/ruby/gems/2.7.0/gems/thor-1.1.0/lib/thor/command.rb:27:in `run'
	from /opt/hostedtoolcache/Ruby/2.7.2/x64/lib/ruby/gems/2.7.0/gems/thor-1.1.0/lib/thor/invocation.rb:127:in `invoke_command'
	from /opt/hostedtoolcache/Ruby/2.7.2/x64/lib/ruby/gems/2.7.0/gems/thor-1.1.0/lib/thor.rb:392:in `dispatch'
	from /opt/hostedtoolcache/Ruby/2.7.2/x64/lib/ruby/gems/2.7.0/gems/thor-1.1.0/lib/thor/base.rb:485:in `start'
	from /opt/hostedtoolcache/Ruby/2.7.2/x64/lib/ruby/gems/2.7.0/gems/linkedin_orbit-0.1.3/bin/linkedin_orbit:26:in `<top (required)>'
	from /opt/hostedtoolcache/Ruby/2.7.2/x64/bin/linkedin_orbit:23:in `load'
	from /opt/hostedtoolcache/Ruby/2.7.2/x64/bin/linkedin_orbit:23:in `<main>'
Error: Process completed with exit code 1.
```

--- a/lib/linkedin_orbit/linkedin.rb
+++ b/lib/linkedin_orbit/linkedin.rb
@@ -12,6 +12,8 @@ module LinkedinOrbit
     def process_comments
       posts = get_posts
 
+      return posts unless posts.is_a?(Array)
+
       posts.each do |post|
         comments = get_post_comments(post["id"])
 
@@ -47,7 +49,16 @@ module LinkedinOrbit
       response = https.request(request)
 
       response = JSON.parse(response.body)
-      
+
+      return response["message"] if response["serviceErrorCode"]
+
+      if response["elements"].nil? || response["elements"].empty?
+        return <<~HEREDOC
+          No new posts to process from your LinkedIn organization.
+          If you suspect this is incorrect, verify your LinkedIn organization schema is correct in your credentials.
+        HEREDOC
+      end
+
       response["elements"].each do |element|
         posts << {
           "id" => element["activity"],

--- a/spec/linkedin_spec.rb
+++ b/spec/linkedin_spec.rb
@@ -1,0 +1,53 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe LinkedinOrbit::Linkedin do
+  let(:subject) do
+    LinkedinOrbit::Linkedin.new(
+      linkedin_organization: "org",
+      linkedin_token: "abc123",
+      orbit_api_key: "12345",
+      orbit_workspace: "1234"
+    )
+  end
+
+  describe "#get_posts" do
+    context "with no posts to process" do
+      it "returns a string message" do
+        stub_request(:get, "https://api.linkedin.com/v2/shares?owners=org&q=owners").
+        with(
+          headers: {
+          'Accept'=>'application/json',
+          'Authorization'=>'Bearer abc123',
+          'Content-Type'=>'application/json',
+          }).
+        to_return(status: 200, body: "{\"elements\": []}", headers: {})
+
+        expect(subject.get_posts).to eql("No new posts to process from your LinkedIn organization.\nIf you suspect this is incorrect, verify your LinkedIn organization schema is correct in your credentials.\n")
+      end
+    end
+
+    context "with posts to process" do
+      it "returns them in the right formatting at the end of the method" do
+        stub_request(:get, "https://api.linkedin.com/v2/shares?owners=org&q=owners").
+        with(
+          headers: {
+          'Accept'=>'application/json',
+          'Authorization'=>'Bearer abc123',
+          'Content-Type'=>'application/json',
+          }).
+        to_return(status: 200, body: "{\"elements\": [{\"owner\": \"org\", \"activity\": \"activity-123\", \"text\": {\"text\": \"LinkedIn Post Body\"}}]}", headers: {})
+        
+        expect(subject.get_posts).to eql(
+          [
+            {
+              "id" => "activity-123",
+              "message_highlight" => "LinkedIn Post Body"
+            }
+          ]
+        )
+      end
+    end
+  end
+end

